### PR TITLE
♻️ dataset_ingest_config to python module

### LINKED ISSUE #166: convert ingest study config yml to a python module
the yaml lib is raising that annoying python 3.8 warning and besides I don't like yaml anyway

### PR DESCRIPTION
**Pull request by fiendish**
convert dataset_ingest_config.yml to dataset_ingest_config.py
closes #166

Will have significant benefits for later TODOs relating to accounting and custom decisions about what to not load

### CODE REVIEW

File: kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py [hunk 1]
```diff
@@ -26,35 +31,48 @@ def __init__(self, config_path):
...
+            assert os.path.isdir(self.extract_config_dir)
+
+            self.study
+            assert_safe_type(self.study, str)
+
+            self.target_service_entities
+            assert_safe_type(self.target_service_entities, list)
+            assert_all_safe_type(self.target_service_entities, str)
+        except (AttributeError, TypeError, AssertionError) as e:
+            raise ConfigValidationError(
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py]: Repetitive, already happens in base_config - https://github.com/kids-first/kf-lib-data-ingest/blob/ae93b8ce2eabb25a1dc94aa74f6d9b4ef5ddd00b/kf_lib_data_ingest/etl/configuration/base_config.py#L28

File: kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py [hunk 2]
```diff
@@ -26,35 +31,48 @@ def __init__(self, config_path):
...
-                                                  filename)
-                                     for filename in os.listdir(
-                                     self.extract_config_dir)
-                                     if filename.endswith('.py')]
+                DATASET_INGEST_CONFIG_DEFAULT_FILENAME
+            )
+
+        super().__init__(config_path)
+
+        # Validate
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py]: This should all be inside the `_validate` method, which is part of the interface of the `base_config.py`. The `_validate` method gets called after the config file is read. See 
https://github.com/kids-first/kf-lib-data-ingest/blob/ae93b8ce2eabb25a1dc94aa74f6d9b4ef5ddd00b/kf_lib_data_ingest/etl/configuration/base_config.py#L26

File: kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py [hunk 3]
```diff
@@ -26,35 +31,48 @@ def __init__(self, config_path):
...
+        try:
+            self.extract_config_dir
+            assert_safe_type(self.extract_config_dir, str)
+
+            # Directory containing the extract config files
+            self.extract_config_dir = os.path.join(
+                os.path.dirname(self.config_filepath),
+                self.extract_config_dir
+            )
+            self.extract_config_paths = [
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py]: This should just be validation. We should separate the setting of config variables or any other business logic from validation logic.

File: kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py [hunk 4]
```diff
@@ -26,35 +31,48 @@ def __init__(self, config_path):
...
-                                     self.extract_config_dir)
-                                     if filename.endswith('.py')]
+                DATASET_INGEST_CONFIG_DEFAULT_FILENAME
+            )
+
+        super().__init__(config_path)
+
+        # Validate
+        try:
+            self.extract_config_dir
```
Comment (znatty22) [inline kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py]: I still don't see anything that tells the user they are missing required attributes..

Comment (fiendish) [reply kf_lib_data_ingest/etl/configuration/dataset_ingest_config.py]: "ERROR: foo is missing" is the same as "ERROR: foo is required and is missing".

**Pull request merged**